### PR TITLE
Prepare GuiRunner to be made private

### PR DESCRIPTION
**Pull request by chapulina**
The `GuiRunner` class was never meant to be installed. This PR starts the tick-tock deprecation of that class so it can be moved to `src/gui` and no longer installed.

* Deprecate the constructor on v5
* Use PIMPL pattern so that the class can be extended during v5 if needed
* Move the header to `src/gui` on v6

### CODE REVIEW

File: include/ignition/gazebo/gui/GuiRunner.hh [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include <ignition/msgs/serialized.pb.h>
 
 #include <QtCore>
+#include <memory>
 #include <string>
 
 #include <ignition/transport/Node.hh>
```
Comment (caguero) [inline include/ignition/gazebo/gui/GuiRunner.hh:26]: You can possibly move this `include` and the next one to the `.cc`.

**Review by caguero — COMMENTED** [review]
Just a minor comment, other than that looks good to me!

File: include/ignition/gazebo/gui/GuiRunner.hh (hunk 1 above)
Comment (chapulina) [reply include/ignition/gazebo/gui/GuiRunner.hh]: True! Moved in 66a2192

**Review by ahcorde — CHANGES_REQUESTED** [review]
UNIT_Gui_TEST is not compiling

```bash
  ../../lib/libignition-gazebo5-gui.so.5.0.0~pre1: undefined reference to `ignition::gazebo::IGNITION_GAZEBO_VERSION_NAMESPACE::GuiRunner::~GuiRunner()'
  ../../lib/libignition-gazebo5-gui.so.5.0.0~pre1: undefined reference to `ignition::gazebo::IGNITION_GAZEBO_VERSION_NAMESPACE::GuiRunner::RequestState()'
  ../../lib/libignition-gazebo5-gui.so.5.0.0~pre1: undefined reference to `ignition::gazebo::IGNITION_GAZEBO_VERSION_NAMESPACE::GuiRunner::~GuiRunner()'
  ../../lib/libignition-gazebo5-gui.so.5.0.0~pre1: undefined reference to `ignition::gazebo::IGNITION_GAZEBO_VERSION_NAMESPACE::GuiRunner::OnPluginAdded(QString const&)'
  ../../lib/libignition-gazebo5-gui.so.5.0.0~pre1: undefined reference to `vtable for ignition::gazebo::v5::GuiRunner'
```

**Comment by chapulina** [comment]
> UNIT_Gui_TEST is not compiling

I think I fixed it in ae049d0

**Comment by ahcorde** [comment]
issues on MACOS https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-homebrew-amd64/4927/consoleFull#2138011276ea37b8d-a6d4-40ae-8431-d90c018842af

```bash
In file included from /Users/jenkins/workspace/ignition_gazebo-ci-pr_any-homebrew-amd64/ign-gazebo/src/gui/GuiRunner.cc:29:
In file included from /Users/jenkins/workspace/ignition_gazebo-ci-pr_any-homebrew-amd64/ign-gazebo/include/ignition/gazebo/gui/GuiRunner.hh:26:
In file included from /usr/local/include/ignition/utils1/ignition/utils/ImplPtr.hh:246:
/usr/local/include/ignition/utils1/ignition/utils/detail/ImplPtr.hh:130:46: error: chosen constructor is explicit in copy-initialization
            new T{std::forward<Args>(args)...},
                                             ^
/Users/jenkins/workspace/ignition_gazebo-ci-pr_any-homebrew-amd64/ign-gazebo/src/gui/GuiRunner.cc:53:20: note: in instantiation of function template specialization 'ignition::utils::MakeUniqueImpl<ignition::gazebo::v5::GuiRunner::Implementation>' requested here
  : dataPtr(utils::MakeUniqueImpl<Implementation>())
                   ^
/usr/local/include/ignition/transport10/ignition/transport/Node.hh:212:24: note: explicit constructor declared here
      public: explicit Node(const NodeOptions &_options = NodeOptions());
                       ^
/Users/jenkins/workspace/ignition_gazebo-ci-pr_any-homebrew-amd64/ign-gazebo/src/gui/GuiRunner.cc:42:27: note: in implicit initialization of field 'node' with omitted initializer
  public: transport::Node node;
                          ^
1 error generated.
make[2]: *** [src/gui/CMakeFiles/ignition-gazebo5-gui.dir/GuiRunner.cc.o] Error 1
```

**Comment by chapulina** [comment]
> issues on MACOS

phew, fixed it by calling `transport::Node`'s constructor.

**Review by ahcorde — APPROVED** [review]
there are some tests failing in the github actions but they are unrelated

**Pull request merged**